apps/api/src/app/change/usecases/promote-notification-template-change/promote-notification-template-change.usecase.ts

```diff
@@ -53,6 +57,30 @@ export class PromoteNotificationTemplateChange {
         return message._parentId === step._templateId;
       });
 
+      if (step.variants && step.variants.length > 0) {
+        step.variants = step.variants
+          ?.map(mapNewVariantItem)
+          .filter((variant): variant is StepVariantEntity => variant !== undefined);
+      }
```

Comment:
question, can you please tell me in which case the variant could be `undefined`? I might be missing something